### DISCUSSION TITLE
CÓDIGOS BIEN HECHOS ENTREGAS

### DISCUSSION THREAD

**Discussion by Daniel-Cotarelo** [mmasias/24-25-PRG1#467] [General]
Hola Manuel, puedes subirnos por aquí los códigos correctos de:
- Cálculo nota final.
- Número aleatorio (Entrega 02-10).
-Piedra papel o tijera.
-Juego del caracol.
Gracias, un saludo.

**Comment by mmasias** [comment]
@Daniel-Cotarelo y todos:

Aquí va una propuesta de código y, para entretenernos un poco una vez entendidos, unos cuantos retos adicionales a desarrollar sobre la solución propuesta (o sobre la que han trabajado, o sobre ambas!)

|Propuesta de código|Retos adicionales
|-|-
|[Cálculo de la nota final](https://github.com/mmasias/24-25-PRG1/blob/main/entregas/masiasManuel/src/CalculoNotaFinal.java)|Agregar la validación de nota para el usuario (que no ponga notas menores que 0 o mayores que 10, y de hacerlo, volverle a preguntar hasta que introduzca una nota válida)
||Agregarle la condición (existente en UNEATLANTICO) de que el examen final hay que superarlo con al menos un 4.0, porque si no la nota final es la misma que la del examen final
||Hecho esto, dígale al usuario si aprobó o no...
|[Piedra, papel & tijeras](https://github.com/mmasias/24-25-PRG1/blob/main/entregas/masiasManuel/src/PiedraPapelTijeras.java)|Agregar la validación de jugada correcta para el usuario (es decir, si elige mal la jugada, volverle a preguntar)
||Agregar un mini-torneo para que resulte ganador el primero que llegue a tres partidas ganadas.
|Juego del caracol [versión 1](https://github.com/mmasias/24-25-PRG1/blob/984635609fa553e0a4ca507d6c537aab4eab24ec/entregas/masiasManuel/src/Caracol.java) y otra [versión full ternario!!!](https://github.com/mmasias/24-25-PRG1/blob/f5be4f309f1a2d4b1a9feff3433470737fa52b90/entregas/masiasManuel/src/Caracol.java)|Agregar la siguiente situación: cada cinco días, toda agua que haya caido en el pozo  se evapora

**Reply by Daniel-Cotarelo** [reply]
Muchas gracias!!

**Comment by Bienvee3** [comment]
Buenas noches Manuel. 
¿Podrías proporcionarnos tu solución del código del juego del caracol?
Gracias, un saludo.

**Reply by mmasias** [reply]
Lo tienes arriba, en la respuesta que le he dado a @Daniel-Cotarelo
